src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -528,6 +543,7 @@ public void setCharAt(int index, char ch) {
             }
             StringUTF16.putCharSB(value, index, ch);
         }
+        growOnly = false;
```

Comment:
Any changes that take place while the string is still latin1 do not make it potentially compressible. You could move this assignment to the `else` clause for some additional possible performance boost. But then the variable name would be misleading.

Reply:
yes, I really hesitate about that. 'growOnly' is an attribute of current AbstractStringBuilder.  if we ignore latin-1 path, it breaks the invariant. Maybe we rename it 'isCompressible'?